components/style/values/specified/pointing.rs

```diff
@@ -0,0 +1,112 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! Specified values for Pointing properties.
+//!
+//! https://drafts.csswg.org/css-ui/#pointing-keyboard
+
+use cssparser::Parser;
+use parser::{Parse, ParserContext};
+use selectors::parser::SelectorParseErrorKind;
+use style_traits::ParseError;
+use style_traits::cursor::CursorKind;
+
+/// The specified value for the `cursor` property.
+///
+/// https://drafts.csswg.org/css-ui/#cursor
+pub use values::computed::pointing::Cursor;
+#[cfg(feature = "gecko")]
+pub use values::computed::pointing::CursorImage;
+#[cfg(feature = "gecko")]
+use values::specified::url::SpecifiedUrl;
+
+impl Cursor {
+    /// Set `cursor` to `auto`
+    #[cfg(not(feature = "gecko"))]
+    #[inline]
+    pub fn auto() -> Self {
+        Self {
+            keyword: CursorKind::Auto,
+        }
+    }
+
+    /// Set `cursor` to `auto`
+    #[cfg(feature = "gecko")]
+    #[inline]
+    pub fn auto() -> Self {
+        Self {
+            images: vec![],
+            keyword: CursorKind::Auto
+        }
+    }
+
+    /// cursor: [auto | default | ...]
+    #[cfg(not(feature = "gecko"))]
+    pub fn parse<'i, 't>(
```

Comment:
This should be a `Parse` trait implementation instead, ditto for the other below.